### PR TITLE
Add documentation on some of the URDF extensions used in the wild

### PR DESCRIPTION
**Pull request by traversaro**
Just an incomplete survey, useful as @xela-95 was recently looking into supporting closed loop robots in some of our libraries, so we also need to understand how to model them.

Fix https://github.com/ami-iit/awesome-urdf/issues/10 .

### CODE REVIEW

File: urdf-extensions.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
+## URDF Extensions
+
+Unfortunately the official URDF specification does not provide any facility to clearly define "extensions" to the URDF, i.e. XML code that is defined inline in URDF XML, but that is not part of the official tags defined in the spec, as for example is done in SDF: http://sdformat.org/tutorials?tut=custom_elements_attributes_proposal&ver=1.7&cat=pose_semantics_docs& . 
+
+However, there are a bunch of URDF parsers that define their own extensions to add some functionalities. What is happening when a URDF parser does not support a given extension, as nothing is specified in the extension, is the extension is simply ignored.
```
Comment (flferretti) [inline urdf-extensions.md:5]:
```suggestion
However, there are a bunch of URDF parsers that define their own extensions to add some functionalities. What is happening when a URDF parser does not support a given extension, as nothing is specified in the extension, is that the extension is simply ignored. 

```

File: README.md [hunk 2]
```diff
@@ -60,13 +61,19 @@ Libraries to import, export and manipulate URDF files.
 - [Foxglove Studio](https://foxglove.dev/urdf) - Visualization app for web or desktop, supports URDFs as well as arbitrary bag or [MCAP](https://mcap.dev) data. [MPL-2.0]
 - [URDFormer](https://github.com/WEIRDLabUW/urdformer) - Given an image, URDFormer predicts its corresponding interactive 'digital twin' in the URDF format. 
 
+### Extensions
+
+Several libraries and groups have extended the URDF format behind the official spec, a brief list is provided in the [`urdf-extensions.md`](urdf-extensions.md) file.
```
Comment (flferretti) [inline README.md:66]:
```suggestion
Several libraries and groups have extended the URDF format beyond the official specs, a brief list is provided in the [`urdf-extensions.md`](urdf-extensions.md) file.
```

File: urdf-extensions.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+### `urdf-ext-mit-biomimetics` 
+
+The Biomimetic Robotics Lab at MIT define its own extensions that they call `URDF+`, that permits to define closed loops in URDF. From the code point of view, they developed a fork on the official libraries `urdfdom_headers` and `urdfdom`.
+
+Documentation: `URDF+` paper at Humanoids 2024, not sure if it is public.
+Code: https://github.com/mit-biomimetics/urdfdom_headers, https://github.com/mit-biomimetics/urdfdom
+
+
+##### Closed loop support
```
Comment (flferretti) [inline urdf-extensions.md]:
```suggestion
#### Closed loop support
```

**Review by flferretti — APPROVED** [review]
Thanks @traversaro, I left some comments

**Review by CarlottaSartore — APPROVED** [review]

**Comment by traversaro** [comment]
Thanks @flferretti !

**Pull request merged**